### PR TITLE
EOS-19896: Factory - Configure storage enclosure

### PR DESCRIPTION
**Pull request by 83bhp**
Signed-off-by: Pritam Bhavsar <pritam.bhavsar@seagate.com>

### CODE REVIEW

File: lr-cli/cortx_setup/commands/storage/commons.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    def fetch_enc_id():
+        # Check if file /etc/enclosure-id has correct content:
+        # 1. has only one line and
+        # 2. has only one word - enclosure serial.
+
+        self.logger.info("Fetching the enclosure ID from {self._enclosure_id_file_path}")       
+        _enc_id_file = Path(self.enclosure_id_file_path)
+        _words = None
+        with open(_enc_id_file) as _fp:
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/commons.py:43]: Seems very complicated logic just to fetch single line value of enclosure-id

File: lr-cli/cortx_setup/commands/storage/controller_config.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        'password': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'password for controller user'
+        },
+        'type': {
+            'type': str,
+            'optional': True,
+            'help': 'Type of controller e.g {Gallium | Indium | virtual}'
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/controller_config.py:56]: Make sure all examples are in small case:
{gallium | indium | virtual}

File: lr-cli/cortx_setup/commands/storage/controller_config.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+from pathlib import Path
+from ..command import Command
+from cortx.utils.conf_store import Conf
+from provisioner.commands import PillarSet
+from provisioner.salt import local_minion_id, function_run
+from .commons import Commons
+
+prvsnr_cluster_path = Path(
+    '/opt/seagate/cortx_configs/provisioner_cluster.json'
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/controller_config.py:26]: Can we have global config which defines Provisioner-Cluster File Path

File: lr-cli/cortx_setup/commands/storage/controller_config.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+    $ cortx_setup storage controller --type (Gallium | Indium | virtual)
+
+    $ cortx_setup storage controller --mode (primary | secondary) --ip <ip-address> --port <port-number>
+    '''
+    _args = {
+        'user': {
+            'type': str,
+            'default': None,
+            'optional': True,
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/controller_config.py:44]: The args are mandatory not optional right???

**Review by nitin-seagate — CHANGES_REQUESTED** [review]

File: lr-cli/cortx_setup/commands/storage/controller_config.py (hunk 2 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/controller_config.py:56]: done

File: lr-cli/cortx_setup/commands/storage/controller_config.py (hunk 4 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/controller_config.py:44]: No, this is optional, as there other options under controller section as well e.g. type/mode/ip/port etc

File: lr-cli/cortx_setup/commands/storage/commons.py (hunk 1 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/commons.py:43]: Will use the logic to read it from grains

File: lr-cli/cortx_setup/commands/storage/commons.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+
+""" Common functions """
+
+class Commons(Command):
+    '''
+    Common functions
+    '''
+    _args = None
+    enclosure_id_file_path = "/etc/enclosure-id"
```
[the remark this answers is not in the document]
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/commons.py:33]: Sure, there's another path in making which will have all the config data consolidated in lr-cli/config.py, once that patch gets landed will add this change in it.

File: lr-cli/cortx_setup/commands/storage/controller_config.py (hunk 3 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/controller_config.py:26]: Sure, there's another path in making which will have all the config data consolidated in lr-cli/config.py, once that patch gets landed will add this change in it.

File: lr-cli/cortx_setup/commands/storage/config.py [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+        },
+        'port': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'Port of the controller to connect to'
+        }
+    }
+
+    def run(self, name=None, type=None,
```
Comment (ypise) [inline lr-cli/cortx_setup/commands/storage/config.py:92]: @mazinamdar , @83bhp 
Can this dest part help?
https://docs.python.org/3/library/argparse.html#the-add-argument-method
```dest - The name of the attribute to be added to the object returned by parse_args().```

**Review by ypise — APPROVED** [review]

File: lr-cli/cortx_setup/commands/storage/config.py (hunk 6 above)
Comment (mazinamdar) [reply lr-cli/cortx_setup/commands/storage/config.py:92]: yes dest will help here ,I tried locally it will work

**Comment by 83bhp** [comment]
Due to lots of conflict issues in this PR the new PR was raised and merged:
https://github.com/Seagate/cortx-prvsnr/pull/1308
Closing this PR.

**Pull request closed without merging**